### PR TITLE
Modifies SortedDL.new for val_res

### PR DESCRIPTION
**Pull request by morganmcg1**
Modifies SortedDL.new to accept val_res, as list of validation set text lengths, to further speed up the init. Also adds additional documentation for SortedDL to emphasise the benefit of passing res to the dataloader

### CODE REVIEW

**Comment by morganmcg1** [comment]
Further elaborated on here: https://forums.fast.ai/t/nlp-speed-up-if-using-sorteddl/74636

**Comment by jph00** [comment]
Thank you!
